### PR TITLE
Enable send lesson dialog [test all browsers]

### PR DESCRIPTION
**Pull request by jamescodeorg**
This should be the final commit for the send lesson dialog.  The only functional change is to move where the button appears in ProgressLessonTeacherInfo and to turn the feature on.  The other changes are all either test-related or to remove code that is no longer needed.

Changes:
- Remove send-lesson-dialog experiment and enable the feature for everyone
- Move send lesson dialog button above the hidden/visible toggle
- Clean up code that is no longer needed
- Add eyes test to verify that dialog renders properly
- Add UI tests to get basic coverage across more browsers

## Links

- [spec](https://docs.google.com/document/d/1k9U-ulkQR2bNZ_BBzRi4PCyjwUAXyn4QzEzdUI4tWbE/edit)
- [jira epic](https://codedotorg.atlassian.net/browse/LP-1574)
- This commit turns on functionality implemented in the following PRs:
    - https://github.com/code-dot-org/code-dot-org/pull/37358
    - https://github.com/code-dot-org/code-dot-org/pull/37429
    - https://github.com/code-dot-org/code-dot-org/pull/37588
    - https://github.com/code-dot-org/code-dot-org/pull/37784

## Testing story

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/progress/ProgressLesson.jsx [hunk 1]
```diff
@@ -153,7 +153,8 @@ class ProgressLesson extends React.Component {
 
     // There's no url for a lesson so use the url of the first level of the lesson
     // as the url for the lesson.
-    // TODO: Make the back-end support lesson urls and redirect to the first level.
+    // TODO: Make the back-end return a lesson url as part of the lesson metadata so we
+    // don't need to pass it separately from lesson here and in ProgressLessonTeacherInfo.
```
Comment (jmkulwik) [inline apps/src/templates/progress/ProgressLesson.jsx:157]: Can you create a jira ticket and link it to a github comment for this TODO.

Comment (jamescodeorg) [reply apps/src/templates/progress/ProgressLesson.jsx:157]: Oh sorry, forgot to include the link here: https://codedotorg.atlassian.net/browse/LP-1634

**Comment by jmkulwik** [comment]
NIT: Most of the work for this was done in a previous PR, right? Could you include a link the PR where the bulk of the work was done (or some way to find it)?

**Review by jmkulwik — APPROVED** [review]
LGTM!

**Pull request merged**